MSALiOSB2C/ViewController.swift

```diff
@@ -351,28 +292,24 @@ class ViewController: UIViewController, UITextFieldDelegate, URLSessionDelegate
     }
     
     override func viewWillAppear(_ animated: Bool) {
-        
-        if self.accessToken.isEmpty {
-            
+        if self.accessToken == nil {
             signoutButton.isEnabled = false;
             callGraphApiButton.isEnabled = false;
             editProfileButton.isEnabled = false;
             refreshTokenButton.isEnabled = false;
-            
-            
         }
     }
     
-    func getUserByPolicy (withUsers: [MSALUser], forPolicy: String) throws -> MSALUser? {
-        
-        for user in withUsers {
-            if (user.userIdentifier().components(separatedBy: ".")[0].hasSuffix(forPolicy.lowercased())) {
-                return user
-            }
+    /**
+     The way B2C knows what actions to perform for the user of the app is through the use of `Authority URL`,
+     a URL indicating a directory that MSAL can use to obtain tokens. In Azure B2C
+     it is of the form `https://<instance/tfp/<tenant>/<policy>`, where `<instance>` is the
+     directory host (e.g. https://login.microsoftonline.com), `<tenant>` is a
+     identifier within the directory itself (e.g. a domain associated to the
+     tenant, such as contoso.onmicrosoft.com), and `<policy>` is the policy you wish to
+     use for the current user flow.
+     */
+    func getAuthority(forPolicy: String) throws -> MSALAuthority {
+        return try MSALAuthority(url: URL(string: String(format: self.kEndpoint, self.kTenantName, forPolicy))!)
```

Comment:
Since it's a B2C sample, we could use MSALB2CAuthority directly as that's the model we'll be recommending going forward.

Reply:
Good idea. I'll introduce a separate PR for this.